py/embedded-server/deephaven_server/__init__.py

```diff
@@ -14,3 +19,55 @@
 # distribution name from the package name.
 # https://docs.python.org/3/library/importlib.metadata.html#package-distributions
 __version__ = importlib.metadata.version('deephaven-server')
+
+
+def main():
+    """
+    Main entry point for the Deephaven embedded server application.
+    Accepts a command to start the application and the arguments to use.
+    Defaults to `start`, which starts the server.
+    """
+    parser = argparse.ArgumentParser(description="Start the Deephaven embedded server")
+    parser.add_argument("cmd", nargs="?", default="start", choices=["start"], help="Deephaven command to run. Defaults to start.")
+    parser.add_argument("-p", "--port", default=8080, type=int, help="The port to bind to. Defaults to 8080.")
+    parser.add_argument("--host", default="localhost", help="The host to bind to. Defaults to localhost.")
+    parser.add_argument("--key", default=None, help="The key to use. Uses a randomly generated key if not specified.")
+    parser.add_argument("--jvm-args", default=None, help="The JVM arguments to use.")
+    parser.add_argument("--dh-args", default=None, help="The Deephaven arguments to use.")
+    args = parser.parse_args()
+
+    if args.cmd == "start":
+        _start(args)
+    else:
+        parser.print_help()
+        sys.exit(1)
+
+def _start(args):
+    """
+    Start the Deephaven server with the given arguments.
+
+    Args:
+        args: The arguments parsed using argparse to use to start the server.
+    """
+    if args.key is None:
+        args.key = secrets.token_urlsafe(32)
+
+    jvm_args = [f"-Dauthentication.psk={args.key}"]
+    if args.jvm_args is not None:
+        jvm_args += args.jvm_args.split()
+
+    s = Server(host=args.host, port=args.port, jvm_args=jvm_args, dh_args=args.dh_args)
```

Comment:
Been a while since I've looked at this layer. It seems like `dh_args` doesn't do anything. We should probably remove it from the stack?

Reply:
Yea they don't do anything, should I remove it from EmbeddedServer as well or just from here?

Reply:
I think so. I would double check w/ @niloc132